File: backend/chat-store-service/src/main/java/com/example/chatstoreservice/chat/repository/ChatRepository.java
package com.example.chatstoreservice.chat.repository;

import com.example.chatstoreservice.chat.entity.Chat;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ChatRepository extends MongoRepository<Chat, String> {

    Optional<Chat> findByChatId(Long chatId);
}

File: backend/chat-store-service/src/main/kotlin/com/example/chatstoreservice/chat/repository/ChatRepository.kt
package com.example.chatstoreservice.chat.repository

import com.example.chatstoreservice.chat.entity.Chat
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface ChatRepository : MongoRepository<Chat, String> {

    fun findByChatId(chatId: Long?): Optional<Chat>
}
